WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,230 @@
+			'jquery-effects-shake'     => true,
+			'jquery-effects-slide'     => true,
+			'jquery-effects-transfer'  => true,
+			'wp-mediaelement'          => true,
+			'schedule'                 => true,
+			'suggest'                  => true,
+			'thickbox'                 => true,
+			'hoverIntent'              => true,
+			'jquery-hotkeys'           => true,
+			'sack'                     => true,
+			'quicktags'                => true,
+			'iris'                     => true,
+			'farbtastic'               => true,
+			'colorpicker'              => true,
+			'tiny_mce'                 => true,
+			'autosave'                 => true,
+			'wp-ajax-response'         => true,
+			'wp-lists'                 => true,
+			'common'                   => true,
+			'editorremov'              => true,
+			'editor-functions'         => true,
+			'ajaxcat'                  => true,
+			'admin-categories'         => true,
+			'admin-tags'               => true,
+			'admin-custom-fields'      => true,
+			'password-strength-meter'  => true,
+			'admin-comments'           => true,
+			'admin-users'              => true,
+			'admin-forms'              => true,
+			'xfn'                      => true,
+			'upload'                   => true,
+			'postbox'                  => true,
+			'slug'                     => true,
+			'post'                     => true,
+			'page'                     => true,
+			'link'                     => true,
+			'comment'                  => true,
+			'comment-reply'            => true,
+			'admin-gallery'            => true,
+			'media-upload'             => true,
+			'admin-widgets'            => true,
+			'word-count'               => true,
+			'theme-preview'            => true,
+			'json2'                    => true,
+			'plupload'                 => true,
+			'plupload-all'             => true,
+			'plupload-html4'           => true,
+			'plupload-html5'           => true,
+			'plupload-flash'           => true,
+			'plupload-silverlight'     => true,
+			'underscore'               => true,
+			'backbone'                 => true,
+			'imagesloaded'             => true,
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.xx.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		if ( ! isset( $parameters[1] ) ) {
+			return;
+		}
+
+		$matched_parameter = $this->strip_quotes( $parameters[1]['raw'] );
+
+		if ( isset( $this->target_functions[ $matched_content ][ $matched_parameter ] ) ) {
+			$this->throw_prohibited_error( $stackPtr, array( $matched_parameter ) );
```

Comment:
It may be better to show the error at the parameter instead of the function. This is not a blocker.

Reply:
A case can be made for either. Is the _deregistering_ the most important bit or is the fact that its a _core script_ being deregistered that's important ?

Reply:
For me the important part of this is that it's deregistering a _core script_. Sometimes scripts could be deregistered (such as child theme removing parent scripts) but core scripts almost always should not be removed.

Reply:
Thanks for the feedback. I've added an additional commit which changes the line on which the error/warning is reported to the line containing the script handle.